### PR TITLE
 Update select placement on commit details page

### PR DESCRIPTION
**Pull request by rohitvinnakota-codecov**
# Description

Similar to https://github.com/codecov/gazebo/pull/2675, updates placement on commit detail pages. This will close out [#836](https://github.com/codecov/engineering-team/issues/836).

# Screenshots

Screenshot 2024-04-02 at 7 05 09 PM
Screenshot 2024-04-02 at 7 05 27 PM

# Link to Sample Entry

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Comment by hootener** [comment]
**Test Failures Detected**: Due to failing tests, we cannot provide coverage reports at this time.

### :x: Failed Test Results: 
Completed 2598 tests with **`5 failed`**, 2593 passed and 0 skipped.
<details><summary>View the full list of failed tests</summary>

| **Test Description** | **Failure message** |
| :-- | :-- |
| <pre>Testsuite:<br>CommitCoverageTabs<br>Test name:<br>CommitCoverageTabs flags multi-select user is not on a team plan renders flag multi-select::CommitCoverageTabs flags multi-select user is not on a team plan renders flag multi-select<br>Envs:<br>- default<br></pre> | <pre>Error: Unable to find an element with the text: All flags. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.<br><br>Ignored nodes: comments, script, style<br>&lt;body&gt;<br>  &lt;div&gt;<br>    &lt;div<br>      class=&quot;mx-0 flex flex-col-reverse justify-between gap-2 border-b border-ds-gray-tertiary md:flex-col xl:flex-row @md/commit-detail-page:!flex-col-reverse @4xl/commit-detail-page:!flex-row&quot;<br>    &gt;<br>      &lt;nav<br>        class=&quot;flex overflow-auto&quot;<br>      &gt;<br>        &lt;a<br>          aria-current=&quot;page&quot;<br>          class=&quot;mr-5 py-2 hover:border-b-4 hover:border-ds-gray-quinary text-ds-gray-quinary whitespace-nowrap !text-ds-gray-octonary border-b-4 !border-ds-gray-octonary font-semibold&quot;<br>          data-cy=&quot;commit&quot;<br>          data-marketing=&quot;commit&quot;<br>          href=&quot;.../cool-repo/commit/sha256&quot;<br>        &gt;<br>          Files changed<br>          &lt;sup<br>            class=&quot;text-xs&quot;<br>          /&gt;<br>        &lt;/a&gt;<br>        &lt;a<br>          class=&quot;mr-5 py-2 hover:border-b-4 hover:border-ds-gray-quinary text-ds-gray-quinary whitespace-nowrap&quot;<br>          data-cy=&quot;commitIndirectChanges&quot;<br>          data-marketing=&quot;commitIndirectChanges&quot;<br>          href=&quot;.../cool-repo/commit/sha256/indirect-changes&quot;<br>        &gt;<br>          Indirect changes<br>          &lt;sup<br>            class=&quot;text-xs&quot;<br>          /&gt;<br>        &lt;/a&gt;<br>        &lt;a<br>          class=&quot;mr-5 py-2 hover:border-b-4 hover:border-ds-gray-quinary text-ds-gray-quinary whitespace-nowrap&quot;<br>          data-cy=&quot;commitTreeView&quot;<br>          data-marketing=&quot;commitTreeView&quot;<br>          href=&quot;.../cool-repo/commit/sha256/tree&quot;<br>        &gt;<br>          File explorer<br>        &lt;/a&gt;<br>      &lt;/nav&gt;<br>    &lt;/div&gt;<br>  &lt;/div&gt;<br>&lt;/body&gt;<br>    at waitForWrapper (.../gazebo/gazebo/node_modules/@.../dom/dist/wait-for.js:162:27)<br>    at findByText (.../gazebo/gazebo/node_modules/@.../dom/dist/query-helpers.js:86:32)<br>    at Object.&lt;anonymous&gt; (.../CommitCoverage/CommitCoverageTabs/CommitCoverageTabs.spec.jsx:343:41)<br>    at Promise.then.completed (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:391:28)<br>    at new Promise (&lt;anonymous&gt;)<br>    at callAsyncCircusFn (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:316:10)<br>    at _callCircusTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:218:40)<br>    at _runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:155:3)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:66:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at run (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:25:3)<br>    at runAndTransformResultsToJestFormat (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)<br>    at jestAdapter (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:68:19)<br>    at runTestInternal (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:389:16)<br>    at runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:475:34)<br>    at Object.worker (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/testWorker.js:133:12)</pre> |
| <pre>Testsuite:<br>CommitCoverageTabs<br>Test name:<br>CommitCoverageTabs flags multi-select user is on a team plan repo is public renders flag multi-select::CommitCoverageTabs flags multi-select user is on a team plan repo is public renders flag multi-select<br>Envs:<br>- default<br></pre> | <pre>Error: Unable to find an element with the text: All flags. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.<br><br>Ignored nodes: comments, script, style<br>&lt;body&gt;<br>  &lt;div&gt;<br>    &lt;div<br>      class=&quot;mx-0 flex flex-col-reverse justify-between gap-2 border-b border-ds-gray-tertiary md:flex-col xl:flex-row @md/commit-detail-page:!flex-col-reverse @4xl/commit-detail-page:!flex-row&quot;<br>    &gt;<br>      &lt;nav<br>        class=&quot;flex overflow-auto&quot;<br>      &gt;<br>        &lt;a<br>          aria-current=&quot;page&quot;<br>          class=&quot;mr-5 py-2 hover:border-b-4 hover:border-ds-gray-quinary text-ds-gray-quinary whitespace-nowrap !text-ds-gray-octonary border-b-4 !border-ds-gray-octonary font-semibold&quot;<br>          data-cy=&quot;commit&quot;<br>          data-marketing=&quot;commit&quot;<br>          href=&quot;.../cool-repo/commit/sha256&quot;<br>        &gt;<br>          Files changed<br>          &lt;sup<br>            class=&quot;text-xs&quot;<br>          /&gt;<br>        &lt;/a&gt;<br>        &lt;a<br>          class=&quot;mr-5 py-2 hover:border-b-4 hover:border-ds-gray-quinary text-ds-gray-quinary whitespace-nowrap&quot;<br>          data-cy=&quot;commitIndirectChanges&quot;<br>          data-marketing=&quot;commitIndirectChanges&quot;<br>          href=&quot;.../cool-repo/commit/sha256/indirect-changes&quot;<br>        &gt;<br>          Indirect changes<br>          &lt;sup<br>            class=&quot;text-xs&quot;<br>          /&gt;<br>        &lt;/a&gt;<br>        &lt;a<br>          class=&quot;mr-5 py-2 hover:border-b-4 hover:border-ds-gray-quinary text-ds-gray-quinary whitespace-nowrap&quot;<br>          data-cy=&quot;commitTreeView&quot;<br>          data-marketing=&quot;commitTreeView&quot;<br>          href=&quot;.../cool-repo/commit/sha256/tree&quot;<br>        &gt;<br>          File explorer<br>        &lt;/a&gt;<br>      &lt;/nav&gt;<br>    &lt;/div&gt;<br>  &lt;/div&gt;<br>&lt;/body&gt;<br>    at waitForWrapper (.../gazebo/gazebo/node_modules/@.../dom/dist/wait-for.js:162:27)<br>    at findByText (.../gazebo/gazebo/node_modules/@.../dom/dist/query-helpers.js:86:32)<br>    at Object.&lt;anonymous&gt; (.../CommitCoverage/CommitCoverageTabs/CommitCoverageTabs.spec.jsx:359:43)<br>    at Promise.then.completed (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:391:28)<br>    at new Promise (&lt;anonymous&gt;)<br>    at callAsyncCircusFn (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:316:10)<br>    at _callCircusTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:218:40)<br>    at _runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:155:3)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:66:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at run (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:25:3)<br>    at runAndTransformResultsToJestFormat (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)<br>    at jestAdapter (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:68:19)<br>    at runTestInternal (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:389:16)<br>    at runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:475:34)<br>    at Object.worker (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/testWorker.js:133:12)</pre> |
| <pre>Testsuite:<br>CommitDetailFileExplorer<br>Test name:<br>CommitDetailFileExplorer rendering table displaying the table head renders components selector::CommitDetailFileExplorer rendering table displaying the table head renders components selector<br>Envs:<br>- default<br></pre> | <pre>Error: Unable to find an element with the text: Components Selector. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.<br><br>Ignored nodes: comments, script, style<br>&lt;body&gt;<br>  &lt;div&gt;<br>    &lt;div<br>      class=&quot;mt-2 flex flex-col gap-2&quot;<br>    &gt;<br>      &lt;div<br>        class=&quot;flex flex-wrap justify-between&quot;<br>      &gt;<br>        &lt;div<br>          class=&quot;flex items-center gap-4&quot;<br>        &gt;<br>          &lt;div<br>            class=&quot;flex items-center gap-4&quot;<br>          &gt;<br>            &lt;div<br>              class=&quot;flex flex-wrap divide-x rounded border&quot;<br>            &gt;<br>              &lt;button<br>                class=&quot;flex-1 py-1 px-2 text-sm cursor-pointer whitespace-nowrap disabled:text-ds-gray-quaternary disabled:border-ds-gray-tertiary disabled:bg-ds-gray-primary bg-ds-primary-base text-white font-semibold rounded-l&quot;<br>              &gt;<br>                Code tree<br>              &lt;/button&gt;<br>              &lt;button<br>                class=&quot;flex-1 py-1 px-2 text-sm cursor-pointer whitespace-nowrap disabled:text-ds-gray-quaternary disabled:border-ds-gray-tertiary disabled:bg-ds-gray-primary rounded-r&quot;<br>              &gt;<br>                File list<br>              &lt;/button&gt;<br>            &lt;/div&gt;<br>          &lt;/div&gt;<br>          &lt;nav<br>            class=&quot;flex flex-1 items-center truncate text-ds-gray-octonary [&amp;&gt;*]:mr-1&quot;<br>          &gt;<br>            &lt;a<br>              class=&quot;<br>  font-sans cursor-pointer<br><br>  hover:underline<br><br>  focus:ring-2<br> text-ds-blue inline-flex items-center gap-1&quot;<br>              data-cy=&quot;commitTreeView&quot;<br>              data-marketing=&quot;commitTreeView&quot;<br>              href=&quot;.../commit/sha256/tree&quot;<br>              text=&quot;cool-repo&quot;<br>            &gt;<br>              cool-repo<br>            &lt;/a&gt;<br>            &lt;span&gt;<br>              /<br>            &lt;/span&gt;<br>            &lt;a<br>              class=&quot;<br>  font-sans cursor-pointer<br><br>  hover:underline<br><br>  focus:ring-2<br> text-ds-blue inline-flex items-center gap-1&quot;<br>              data-cy=&quot;commitTreeView&quot;<br>              data-marketing=&quot;commitTreeView&quot;<br>              href=&quot;.../commit/sha256/tree/a&quot;<br>              text=&quot;a&quot;<br>            &gt;<br>              a<br>            &lt;/a&gt;<br>            &lt;span&gt;<br>              /<br>            &lt;/span&gt;<br>            &lt;a<br>              class=&quot;<br>  font-sans cursor-pointer<br><br>  hover:underline<br><br>  focus:ring-2<br> text-ds-blue inline-flex items-center gap-1&quot;<br>              data-cy=&quot;commitTreeView&quot;<br>              data-marketing=&quot;commitTreeView&quot;<br>              href=&quot;.../commit/sha256/tree/a/b&quot;<br>              text=&quot;b&quot;<br>            &gt;<br>              b<br>            &lt;/a&gt;<br>            &lt;span&gt;<br>              /<br>            &lt;/span&gt;<br>            &lt;span<br>              class=&quot;flex items-center font-semibold&quot;<br>            &gt;<br>              c<br>            &lt;/span&gt;<br>          &lt;/nav&gt;<br>        &lt;/div&gt;<br>        &lt;div<br>          class=&quot;flex gap-2&quot;<br>        &gt;<br>          &lt;div&gt;<br>            &lt;label<br>              class=&quot;block font-semibold mb-2 sr-only&quot;<br>              for=&quot;text-input81&quot;<br>            &gt;<br>              Search for files<br>            &lt;/label&gt;<br>            &lt;div<br>              class=&quot;relative&quot;<br>            &gt;<br>              &lt;div<br>                class=&quot;absolute text-ds-gray-quaternary left-2 h-full flex items-center&quot;<br>              &gt;<br>                &lt;svg<br>                  class=&quot;w-4 h-4&quot;<br>                  data-icon=&quot;&quot;<br>                  data-testid=&quot;&quot;<br>                &gt;<br>                  search.svg<br>                &lt;/svg&gt;<br>              &lt;/div&gt;<br>              &lt;input<br>                class=&quot;block border-ds-gray-tertiary px-3 text-sm w-full h-8 disabled:text-ds-gray-quaternary disabled:bg-ds-gray-primary disabled:border-ds-gray-tertiary rounded border pl-7&quot;<br>                data-marketing=&quot;commit-files-search&quot;<br>                id=&quot;text-input81&quot;<br>                placeholder=&quot;Search for files&quot;<br>                type=&quot;text&quot;<br>                value=&quot;&quot;<br>              /&gt;<br>            &lt;/div&gt;<br>          &lt;/div&gt;<br>        &lt;/div&gt;<br>      &lt;/div&gt;<br>      &lt;div<br>        class=&quot;border-t border-ds-gray-tertiary&quot;<br>      &gt;<br>        &lt;div<br>          class=&quot;flex w-full flex-1 flex-wrap items-start gap-2 bg-white sm:flex-row sm:items-center md:mb-1 lg:w-auto lg:flex-none border-b-0 pb-1 border-ds-gray-tertiary&quot;<br>        &gt;<br>          &lt;div<br>            class=&quot;flex gap-2 pt-2&quot;<br>          &gt;<br>            &lt;div<br>              class=&quot;px-2 capitalize flex text-xs font-mono items-center gap-2 bg-ds-coverage-uncovered border-ds-primary-red border-r-2 font-bold&quot;<br>            &gt;<br>              &lt;span<br>                class=&quot;text-black&quot;<br>              &gt;<br>                uncovered<br>              &lt;/span&gt;<br>              &lt;div<br>                class=&quot;flex items-center&quot;<br>              &gt;<br>                &lt;div<br>                  class=&quot;pr-1 flex items-center text-ds-primary-red&quot;<br>                &gt;<br>                  &lt;svg<br>                    class=&quot;w-4 h-4&quot;<br>                    data-icon=&quot;&quot;<br>                    data-testid=&quot;&quot;<br>                  &gt;<br>                    exclamation-triangle.svg<br>                  &lt;/svg&gt;<br>                &lt;/div&gt;<br>              &lt;/div&gt;<br>            &lt;/div&gt;<br>            &lt;div<br>              class=&quot;px-2 capitalize flex text-xs font-mono items-center gap-2 bg-ds-coverage-partial border-ds-primary-yellow border-r-2 border-dotted font-bold&quot;<br>            &gt;<br>              &lt;span<br>                class=&quot;text-black&quot;<br>              &gt;<br>                p...<br>    at waitForWrapper (.../gazebo/gazebo/node_modules/@.../dom/dist/wait-for.js:162:27)<br>    at findByText (.../gazebo/gazebo/node_modules/@.../dom/dist/query-helpers.js:86:32)<br>    at Object.&lt;anonymous&gt; (.../routes/CommitDetailFileExplorer/CommitDetailFileExplorer.spec.jsx:226:36)<br>    at Promise.then.completed (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:391:28)<br>    at new Promise (&lt;anonymous&gt;)<br>    at callAsyncCircusFn (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:316:10)<br>    at _callCircusTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:218:40)<br>    at runNextTicks (node:internal/process/task_queues:60:5)<br>    at listOnTimeout (node:internal/timers:540:9)<br>    at processTimers (node:internal/timers:514:7)<br>    at _runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:155:3)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:66:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at run (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:25:3)<br>    at runAndTransformResultsToJestFormat (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)<br>    at jestAdapter (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:68:19)<br>    at runTestInternal (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:389:16)<br>    at runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:475:34)<br>    at Object.worker (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/testWorker.js:133:12)</pre> |
| <pre>Testsuite:<br>CommitDetailFileViewer<br>Test name:<br>CommitDetailFileViewer rendering component renders ComponentsSelector::CommitDetailFileViewer rendering component renders ComponentsSelector<br>Envs:<br>- default<br></pre> | <pre>Error: Unable to find an element with the text: ComponentsSelector. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.<br><br>Ignored nodes: comments, script, style<br>&lt;body&gt;<br>  &lt;div&gt;<br>    &lt;div<br>      class=&quot;flex w-full flex-1 flex-wrap items-start gap-2 bg-white sm:flex-row sm:items-center md:mb-1 lg:w-auto lg:flex-none border-b pb-2 border-ds-gray-tertiary&quot;<br>    &gt;<br>      &lt;div<br>        class=&quot;flex gap-2 pt-2&quot;<br>      &gt;<br>        &lt;div<br>          class=&quot;px-2 capitalize flex text-xs font-mono items-center gap-2 bg-ds-coverage-uncovered border-ds-primary-red border-r-2 font-bold&quot;<br>        &gt;<br>          &lt;span<br>            class=&quot;text-black&quot;<br>          &gt;<br>            uncovered<br>          &lt;/span&gt;<br>          &lt;div<br>            class=&quot;flex items-center&quot;<br>          &gt;<br>            &lt;div<br>              class=&quot;pr-1 flex items-center text-ds-primary-red&quot;<br>            &gt;<br>              &lt;svg<br>                class=&quot;w-4 h-4&quot;<br>                data-icon=&quot;&quot;<br>                data-testid=&quot;&quot;<br>              &gt;<br>                exclamation-triangle.svg<br>              &lt;/svg&gt;<br>            &lt;/div&gt;<br>          &lt;/div&gt;<br>        &lt;/div&gt;<br>        &lt;div<br>          class=&quot;px-2 capitalize flex text-xs font-mono items-center gap-2 bg-ds-coverage-partial border-ds-primary-yellow border-r-2 border-dotted font-bold&quot;<br>        &gt;<br>          &lt;span<br>            class=&quot;text-black&quot;<br>          &gt;<br>            partial<br>          &lt;/span&gt;<br>          &lt;div<br>            class=&quot;flex items-center&quot;<br>          &gt;<br>            &lt;div<br>              class=&quot;pr-1 flex items-center text-ds-primary-yellow&quot;<br>            &gt;<br>              &lt;span<br>                data-testid=&quot;partial-icon&quot;<br>              &gt;<br>                !<br>              &lt;/span&gt;<br>            &lt;/div&gt;<br>          &lt;/div&gt;<br>        &lt;/div&gt;<br>        &lt;div<br>          class=&quot;px-2 capitalize flex text-xs font-mono items-center gap-2 bg-ds-coverage-covered border-ds-primary-green font-regular&quot;<br>        &gt;<br>          &lt;span<br>            class=&quot;text-black&quot;<br>          &gt;<br>            covered<br>          &lt;/span&gt;<br>          &lt;div<br>            class=&quot;flex items-center&quot;<br>          /&gt;<br>        &lt;/div&gt;<br>      &lt;/div&gt;<br>      &lt;div<br>        class=&quot;ml-auto flex w-full flex-wrap items-center justify-between gap-2 md:mt-2 md:w-auto&quot;<br>      &gt;<br>        &lt;div<br>          class=&quot;w-full sm:w-60&quot;<br>        &gt;<br>          &lt;div<br>            class=&quot;relative&quot;<br>          &gt;<br>            &lt;div&gt;<br>              &lt;button<br>                aria-controls=&quot;downshift-16-menu&quot;<br>                aria-expanded=&quot;false&quot;<br>                aria-label=&quot;Filter by flags&quot;<br>                class=&quot;flex justify-between items-center w-full border border-ds-gray-tertiary rounded bg-white text-left px-3 h-8 disabled:text-ds-gray-quaternary disabled:bg-ds-gray-primary disabled:border-ds-gray-tertiary focus:outline-1 whitespace-nowrap&quot;<br>                data-marketing=&quot;fileviwer-filter-by-flags&quot;<br>                disabled=&quot;&quot;<br>                id=&quot;downshift-16-toggle-button&quot;<br>                tabindex=&quot;-1&quot;<br>              &gt;<br>                &lt;span<br>                  class=&quot;flex items-center gap-2&quot;<br>                &gt;<br>                  &lt;svg<br>                    class=&quot;w-6 h-6&quot;<br>                    data-icon=&quot;&quot;<br>                    data-testid=&quot;&quot;<br>                  &gt;<br>                    flag.svg<br>                  &lt;/svg&gt;<br>                  All flags<br>                &lt;/span&gt;<br>                &lt;svg<br>                  class=&quot;w-6 h-6&quot;<br>                  data-icon=&quot;&quot;<br>                  data-testid=&quot;&quot;<br>                &gt;<br>                  chevron-down.svg<br>                &lt;/svg&gt;<br>              &lt;/button&gt;<br>              &lt;div<br>                class=&quot;absolute inset-x-0 z-10&quot;<br>              &gt;<br>                &lt;div<br>                  class=&quot;hidden&quot;<br>                &gt;<br>                  &lt;div&gt;<br>                    &lt;label<br>                      class=&quot;block font-semibold mb-2 sr-only&quot;<br>                      for=&quot;text-input21&quot;<br>                    &gt;<br>                      Search for Flags<br>                    &lt;/label&gt;<br>                    &lt;div<br>                      class=&quot;relative&quot;<br>                    &gt;<br>                      &lt;div<br>                        class=&quot;absolute text-ds-gray-quaternary left-2 h-full flex items-center&quot;<br>                      &gt;<br>                        &lt;svg<br>                          class=&quot;w-4 h-4&quot;<br>                          data-icon=&quot;&quot;<br>                          data-testid=&quot;&quot;<br>                        &gt;<br>                          search.svg<br>                        &lt;/svg&gt;<br>                      &lt;/div&gt;<br>                      &lt;input<br>                        aria-activedescendant=&quot;&quot;<br>                        aria-autocomplete=&quot;list&quot;<br>                        aria-controls=&quot;downshift-16-menu&quot;<br>                        aria-expanded=&quot;false&quot;<br>                        aria-labelledby=&quot;downshift-16-label&quot;<br>                        autocomplete=&quot;off&quot;<br>                        class=&quot;block border-ds-gray-tertiary px-3 text-sm w-full h-8 disabled:text-ds-gray-quaternary disabled:bg-ds-gray-primary disabled:border-ds-gray-tertiary border-t border-r border-l focus:border rounded-tl rounded-tr pl-7&quot;<br>                        data-marketing=&quot;multiselect-search&quot;<br>                        id=&quot;downshift-16-input&quot;<br>                        placeholder=&quot;Search for Flags&quot;<br>                        role=&quot;combobox&quot;<br>                        type...<br>    at waitForWrapper (.../gazebo/gazebo/node_modules/@.../dom/dist/wait-for.js:162:27)<br>    at findByText (.../gazebo/gazebo/node_modules/@.../dom/dist/query-helpers.js:86:32)<br>    at Object.&lt;anonymous&gt; (.../routes/CommitDetailFileViewer/CommitDetailFileViewer.spec.jsx:156:37)<br>    at Promise.then.completed (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:391:28)<br>    at new Promise (&lt;anonymous&gt;)<br>    at callAsyncCircusFn (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:316:10)<br>    at _callCircusTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:218:40)<br>    at runNextTicks (node:internal/process/task_queues:60:5)<br>    at listOnTimeout (node:internal/timers:540:9)<br>    at processTimers (node:internal/timers:514:7)<br>    at _runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:155:3)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:66:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at run (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:25:3)<br>    at runAndTransformResultsToJestFormat (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)<br>    at jestAdapter (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:68:19)<br>    at runTestInternal (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:389:16)<br>    at runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:475:34)<br>    at Object.worker (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/testWorker.js:133:12)</pre> |
| <pre>Testsuite:<br>IndirectChangesTab<br>Test name:<br>IndirectChangesTab renders components selector::IndirectChangesTab renders components selector<br>Envs:<br>- default<br></pre> | <pre>Error: Unable to find an element with the text: Components Selector. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.<br><br>Ignored nodes: comments, script, style<br>&lt;body&gt;<br>  &lt;div&gt;<br>    IndirectChangesTable<br>  &lt;/div&gt;<br>&lt;/body&gt;<br>    at waitForWrapper (.../gazebo/gazebo/node_modules/@.../dom/dist/wait-for.js:162:27)<br>    at findByText (.../gazebo/gazebo/node_modules/@.../dom/dist/query-helpers.js:86:32)<br>    at Object.&lt;anonymous&gt; (.../routes/IndirectChangesTab/IndirectChangesTab.spec.jsx:22:35)<br>    at Promise.then.completed (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:391:28)<br>    at new Promise (&lt;anonymous&gt;)<br>    at callAsyncCircusFn (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/utils.js:316:10)<br>    at _callCircusTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:218:40)<br>    at _runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:155:3)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:66:9)<br>    at _runTestsForDescribeBlock (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:60:9)<br>    at run (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/run.js:25:3)<br>    at runAndTransformResultsToJestFormat (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:170:21)<br>    at jestAdapter (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:68:19)<br>    at runTestInternal (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:389:16)<br>    at runTest (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/runTest.js:475:34)<br>    at Object.worker (.../gazebo/gazebo/node_modules/react-scripts/node_modules/jest-runner/build/testWorker.js:133:12)</pre> |

**Comment by hootener** [comment]
## [Bundle](https://stage-app.codecov.dev/gh/codecov/gazebo/pull/2738?dropdown=bundle) Report

Changes will decrease total bundle size by 768 bytes :arrow_down:

| Bundle name | Size | Change |
| ----------- | ---- | ------ |
| gazebo-staging-array-push | 6.05MB | 768 bytes :arrow_down: |

File: src/ui/FileViewer/ToggleHeader/Title/Title.jsx [hunk 1]
```diff
@@ -81,8 +81,10 @@ export const TitleFlags = () => {
     return null
   }
 
+  const selectorWidth = commitDetailView ? 'sm:w-48' : 'sm:w-60'
+
   return (
-    <div className="w-full sm:w-60">
+    <div className={`w-full ${selectorWidth}`}>
```
Comment (RulaKhaled) [inline src/ui/FileViewer/ToggleHeader/Title/Title.jsx:87]: let's use cs from `classnames` to conditionally set styles

**Comment by RulaKhaled** [comment]
code looks good, 2 things though:
1. Selector label looks weird (overflowing the selector box) can you use break-word or anything similar? 
Screenshot 2024-04-03 at 4 14 58 PM


2. The selector goes away in indirect changes tab if there are no files with selected component, can you look into this please?
Screenshot 2024-04-03 at 4 15 30 PM

**Review by RulaKhaled — APPROVED** [review]
lgtm

**Pull request merged**